x-pack/plugins/ml/public/application/trained_models/models_management/models_list.tsx

```diff
@@ -361,12 +401,15 @@ export const ModelsList: FC = () => {
       description: i18n.translate('xpack.ml.inference.modelsList.startModelAllocationActionLabel', {
         defaultMessage: 'Start allocation',
       }),
-      icon: 'download',
+      icon: 'play',
       type: 'icon',
       isPrimary: true,
+      enabled: (item) =>
+        !isLoading && !['started', 'starting'].includes(item.stats?.deployment_stats?.state ?? ''),
```

Comment:
are there constants that can be used for `started` and  `starting`?

Reply:
What about introducing a union type instead? `started` | `starting` | `stoping` etc

Reply:
I'm not sure how a union type could be used for this check, but yes, we probably should have one for `state` in `TrainedModelDeploymentStatsResponse`
Or are you thinking a union type based on the keys on an const or enum or something like that?

Reply:
changed in 33362671dfe75211dc3032ada4afd7ca60db432b